hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java

```diff
@@ -130,8 +130,7 @@ RegionStateNode createRegionStateNode(RegionInfo regionInfo) {
     synchronized (regionsMapLock) {
       RegionStateNode node = regionsMap.computeIfAbsent(regionInfo.getRegionName(),
         key -> new RegionStateNode(regionInfo, regionInTransition));
-      encodedRegionsMap.putIfAbsent(regionInfo.getEncodedName(), node);
-      return node;
+      return encodedRegionsMap.putIfAbsent(regionInfo.getEncodedName(), node);
```

Comment:
This is not correct, if there is no value yet putIfAbsent will returns null...

And here I think we could add an assert, that putIfAbsent should always return null, and then return the node above.

Reply:
How about using `put` rather than `putIfAbsent`?
Because the value of `encodedRegionsMap` must always be the same as the `valuesMap`.
I used `putIfAbsent` because I didn't want to put it all the time when encodedRegionsMap already had a value.
I think compare the value of `encodedRegionsMap` and `node` and it to put only when different.

```java
RegionStateNode node = regionsMap.computeIfAbsent(regionInfo.getRegionName(),
  key -> new RegionStateNode(regionInfo, regionInTransition));
if (encodedRegionsMap.get(regionInfo.getEncodedName()) != node) {
  encodedRegionsMap.put(regionInfo.getEncodedName(), node);
}
return node;
```

I'm sorry to bother you.

Reply:
I'm fine with using put, as there is a lock here...